ee/WFLY-1160-Provide_ability_to_configure_a_global_directory_which_puts_the_contents_in_the_classpath.adoc

```diff
@@ -0,0 +1,94 @@
+
+  [standalone@localhost:9990 /] /subsystem=ee:list-add(name=global-folders,value={name=my-common-libs, path=/Users/me/libs})
+
+  [domain@localhost:9990 /] /profile=default/subsystem=ee:list-add(name=global-folders,value={name=my-common-libs, path=/Users/me/libs})
+
+The name and path attributes will be mandatory.
+Internally, this operation will create a module with the name `my-common-libs` that will have as resource-roots the folder `/Users/me/libs/` and all the jars files included in this folder. Subdirectories will not be scanned, but if it is required, users can create a new global-folder using the subdirectory path.
+
+The module created from the shared folder will be loaded as soon as the first application is deployed in the server. That means, if the server is started/restarted and there are no applications deployed, then the shared folder is neither scanned nor the module loaded.
+
+
+Removal of a global-folder could be done using the following operation:
+
+  [standalone@localhost:9990 /] /subsystem=ee:list-remove(name=global-folders,value={name=my-common-libs})
+
+  [domain@localhost:9990 /] /profile=default/subsystem=ee:list-remove(name=global-folders,value={name=my-common-libs})
+
+This RFE will not cover the availability of hot JBoss modules redeployments, that means, modifications of existing jar files as well as adding a new jar or removal an existing one will require a complete server restart, however, changes in plain resources as property files will be available immediately to the applications.
+
+== Issue Metadata
+
+=== Issue
+* https://issues.jboss.org/browse/EAP7-1124[EAP7-1124]
+  * https://issues.jboss.org/browse/WFLY-1160[WFLY-1160]
+* https://issues.jboss.org/browse/WFCORE-4375[WFCORE-4375]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/MODULES-99[MODULES-99]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:vmarek@redhat.com[Vratislav Marek]
+
+
+=== Testing By
+[ ] Engineering
+
+[X] QE
+
+=== Affected Projects or Components
+Wildfly
+Wildfly-core
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+- The new module created from the global-folder configuration will get automatically a dependency on `javaee.api` module name.
+- Hot JBoss modules redeployments will require a server restart to make them available to all existing and new applications.
+- New modules created from the content of the shared folder will be added to the deployed applications as a system dependencies.
+
+=== Nice-to-Have Requirements
+- We could have a deployment pattern name to filter which deployment is affected by this global shared folder.
+- We could allow the possibility of adding existing module dependencies to the new module created from the global-folder configuration, and, if there is no dependencies configured, use by default `javaee.api`.
+
+=== Non-Requirements
```

Comment:
Please add an item here stating that putting the module content on the filesystem of any machine is not a requirement; it's completely a user responsibility.

There's some question in the comments here that seem to be about whether we're managing the module content.

Reply:
Fixed. Added a non-requirement following your description.